source/index.html.md

```diff
@@ -1005,6 +1005,111 @@ supported.
   Also, <code>initiator</code> cannot be <code>merchant</code>.
 </p>
 
+##### Method: `token`
+
+Mastercard and Visa supply and require different sets of parameters for MDES and
+VTS transactions. Only one scheme can be used for a given authorization.
+
+<dl class="dl-vertical">
+  <dt>
+    token[visa]
+    <span class="type">dictionary</span>
+  </dt>
+  <dd>
+    Visa token data.
+    <div class="type">Optional. Cannot be present if <code>[mastercard]</code> is present.</div>
+  </dd>
+  <dt>
+    token[mastercard]
+    <span class="type">dictionary</span>
+  </dt>
+  <dd>
+    Mastercard token data.
+    <div class="type">Optional. Cannot be present if <code>[visa]</code> is present.</div>
+  </dd>
+</dl>
+
+##### Method: `token[mastercard]`
+
+<dl class="dl-vertical">
+  <dt>token[mastercard][tan]
+    <span class="type">[0-9]{12,19}</span>
+  </dt>
+  <dd>
+    Token Account Number to charge.
+    Found in <code>encryptedPayload.encryptedData.accountNumber</code> in the MDES response.
+  </dd>
+  <dt>token[mastercard][expiry]
+    <span class="type">[0-9]{6}</span>
+  </dt>
+  <dd>
+    Expiry.
+    Found in <code>encryptedPayload.encryptedData.applicationExpiryDate</code> in the MDES response.
+  </dd>
```

Comment:
```suggestion
  <dt>token[mastercard][expire][month]
    <span class="type">[0-9]{2}</span>
  </dt>
  <dd>
    Expire month.
    Found in <code>encryptedPayload.encryptedData.applicationExpiryDate</code> in the MDES response.
  </dd>
  <dt>token[mastercard][expire][year]
    <span class="type">[0-9]{4}</span>
  </dt>
  <dd>
    Expire year.
    Found in <code>encryptedPayload.encryptedData.applicationExpiryDate</code> in the MDES response.
  </dd>
```

Reply:
See my comment about Mastercard SCOF: https://github.com/clearhaus/gateway-api-docs/pull/149#issuecomment-1210492342

SCOF has `MM` and `YYYY`.

However, just like I've mentioned in a parallel comment thread, I don't think we want to have both `[expire_something]` and `[expire][something]` in the API. We already use the former in existing payment methods.

Reply:
This is outdated now so I'll close the thread.